cmd/conformance/aws/main.go

```diff
@@ -138,8 +144,27 @@ func storageConfigFromFlags() aws.Config {
 		*dbUser, *dbPassword, dbEndpoint, *dbName,
 	)
 
+	// Configure to use MinIO Server
+	var awsConfig *aaws.Config
+	var s3Opts func(o *s3.Options)
+	if *s3Endpoint != "" {
+		const defaultRegion = "us-east-1"
```

Comment:
It looks to me that this will always set the region to `us-east-1` as soon as someone sets s3Endpoint. That will then set `SDKConfig` which means that `LoadDefaultConfig` will be skipped, thus skipping other values that would be in the default config. At the moment, we only use `LoadDefaultConfig` for S3 but that might not always be the case, so let's disambiguate this to avoid leaving something to trip on later? Note,  that I don't know how `LoadDefaultConfig` works with MinIO, but it looks like this might be tricky if someone uses both AWS and the new flags. That should not be done in the first place, but if the code doesn't prevent it (should it?!), then let's make sure there's no abiguity?

Reply:
That's intentional - the idea here is that you only ever set `--s3_endpoint` if you're going "off-piste" and running outside of AWS, in which case the aws default config no longer applies.

Reply:
Things I'm thinking of:
  - If someone uses a local MySQL instance, they're "off piste", but might not experience the same behavior depending on whether they specify the S3 endpoint using the flags, or `LoadDefaultConfig`, which is..... fine because they're off piste?
 - Right now `LoadDefaultConfig` is only called for S3, but my plan was to change this such that it can use IAM authentication for MySQL instead of user and password. We might have to reorganise things then, but seems fine to me with the current code.

Reply:
Cool, yeah - I _think_ this is only a problem if they want to use "half AWS, half off"(?) which is definitely a non-supported config, so, as you say, probably ok that it doesn't work.

I'm trying to avoid having the amazon SDK "inadvertently" populate the client with AWS credentials, endpoint URLs, or anything else it does "by default" if you're in off-piste mode.